components/insight/SRC/org/openmicroscopy/shoola/agents/measurement/view/ROITable.java

```diff
@@ -849,25 +811,36 @@ void handleParentFolderNodes(ROINode node) {
         }
     }
     
-    /** 
-     * Find the {@link ROINode} for a certain {@link FolderData} within a
-     * collection of {@link ROINode}s
+    private ROINode findFolderNode(FolderData folder) {
+        Collection<ROINode> tmp = nodesMap.get(getUUID(folder));
+        if (tmp.size() == 1)
+            return tmp.iterator().next();
+        if (CollectionUtils.isEmpty(tmp))
```

Comment:
we already know it's not null because we called size() on it, I'm wondering if you need to either switch these ifs (if it could be null) or just do a switch(tmp.size()) ...